### DISCUSSION TITLE
yarn android is failed!!!please help me

### DISCUSSION THREAD

**Discussion by Ccc437647** [NervJS/taro#15077] [Q & A]
我按照该文档
https://taro.redwoodjs.cn/docs/react-native#%E7%AE%80%E4%BB%8B
创建了一个taro+react native模版项目,
该项目执行yarn dev:rn, yarn:dev:h5, yarn dev:weapp 都成功
然后我接下来想执行 yarn android  一直都是失败状态
( 
 我尝试了以下
使用android studio 创建了一个 android studio 工程项目,  安卓手机使用USB数据线连接电脑, 点击run按钮,会自动安装软件到手机上
创建react native项目,JDK版本为17,安卓手机使用USB数据线连接电脑,执行yarn android,也能成功自动安装软件到手机上
但是,taro+react native模版项目
)
以下是vscode编辑器的报错:
`
PS E:\textCode\myApp> yarn android        
yarn run v1.22.21
$ react-native run-android
info Starting JS server...
info Installing the app...

> Task :react-native-gradle-plugin:compileKotlin
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (10, 37): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (119, 30): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (135, 26): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (157, 32): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (163, 31): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactExtension.kt: (171, 36): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\ReactPlugin.kt: (114, 48): 'reactRoot: DirectoryProperty' is deprecated. reactRoot was confusing and has been replace with root to point to your root project and reactNativeDir to point to the folder of the react-native 
NPM package
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (10, 37): 'ApplicationVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (11, 37): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (12, 37): 'LibraryVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (27, 51): 'BaseVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (130, 12): 'ApplicationVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (131, 12): 'LibraryVariant' is deprecated. Deprecated in Java
w: E:\textCode\myApp\node_modules\react-native-gradle-plugin\src\main\kotlin\com\facebook\react\TaskConfiguration.kt: (251, 14): 'BaseVariant' is deprecated. 
Deprecated in Java

> Configure project :expo-application
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-av
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
5 actionable tasks: 5 executed

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script 'E:\textCode\myApp\node_modules\expo-modules-autolinking\scripts\android\autolinking_implementation.gradle' line: 272

* What went wrong:
A problem occurred evaluating project ':expo'.
> A problem occurred configuring project ':expo-av'.
   > Failed to notify project evaluation listener.
      > Could not determine NDK host architecture. None of the following directories exist: D:\Download\AndroidSDK\ndk\21.4.7075529\prebuilt\windows-x86_64,D:\Download\AndroidSDK\ndk\21.4.7075529\prebuilt\windows
      > Could not get unknown property 'release' for SoftwareComponentInternal set of type org.gradle.api.internal.component.DefaultSoftwareComponentContainer.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':expo'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 56s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script 'E:\textCode\myApp\node_modules\expo-modules-autolinking\scripts\android\autolinking_implementation.gradle' line: 272

* What went wrong:
A problem occurred evaluating project ':expo'.
> A problem occurred configuring project ':expo-av'.
   > Failed to notify project evaluation listener.
      > Could not determine NDK host architecture. None of the following directories exist: D:\Download\AndroidSDK\ndk\21.4.7075529\prebuilt\windows-x86_64,D:\Download\AndroidSDK\ndk\21.4.7075529\prebuilt\windows
      > Could not get unknown property 'release' for SoftwareComponentInternal set of type org.gradle.api.internal.component.DefaultSoftwareComponentContainer.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':expo'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 56s

    at makeError (E:\textCode\myApp\node_modules\execa\index.js:174:9)
    at E:\textCode\myApp\node_modules\execa\index.js:278:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runOnAllDevices (E:\textCode\myApp\node_modules\@react-native-community\cli-platform-android\build\commands\runAndroid\runOnAllDevices.js:109:5) 
    at async Command.handleAction (E:\textCode\myApp\node_modules\@react-native-community\cli\build\index.js:142:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`

**Comment by Ccc437647** [comment]
这是项目的package.json文件
{
  "name": "myApp",
  "version": "1.0.0",
  "private": true,
  "description": "hello",
  "templateInfo": {
    "name": "react-native",
    "typescript": true,
    "css": "Sass",
    "framework": "React"
  },
  "scripts": {
    "build:weapp": "taro build --type weapp",
    "build:swan": "taro build --type swan",
    "build:alipay": "taro build --type alipay",
    "build:tt": "taro build --type tt",
    "build:h5": "taro build --type h5",
    "build:rn": "taro build --type rn",
    "build:qq": "taro build --type qq",
    "build:jd": "taro build --type jd",
    "build:quickapp": "taro build --type quickapp",
    "dev:weapp": "npm run build:weapp -- --watch",
    "dev:swan": "npm run build:swan -- --watch",
    "dev:alipay": "npm run build:alipay -- --watch",
    "dev:tt": "npm run build:tt -- --watch",
    "dev:h5": "npm run build:h5 -- --watch",
    "dev:rn": "npm run build:rn -- --watch --qr",
    "dev:qq": "npm run build:qq -- --watch",
    "dev:jd": "npm run build:jd -- --watch",
    "dev:quickapp": "npm run build:quickapp -- --watch",
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "podInstall": "pod-install",
    "upgradePeerdeps": "install-peerdeps @tarojs/taro-rn@3.6.22 -o -Y && install-peerdeps @tarojs/components-rn@3.6.22 -o -Y && install-peerdeps @tarojs/router-rn@3.6.22 -o -Y && pod-install"
  },
  "browserslist": [
    "last 3 versions",
    "Android >= 4.1",
    "ios >= 8"
  ],
  "author": "",
  "dependencies": {
    "@babel/runtime": "^7.7.7",
    "@bam.tech/react-native-image-resizer": "~3.0.4",
    "@react-native-async-storage/async-storage": "~1.17.7",
    "@react-native-camera-roll/camera-roll": "~5.0.4",
    "@react-native-clipboard/clipboard": "~1.11.1",
    "@react-native-community/geolocation": "~3.0.2",
    "@react-native-community/netinfo": "~9.3.0",
    "@react-native-community/slider": "~4.3.2",
    "@react-native-masked-view/masked-view": "~0.2.7",
    "@react-native-picker/picker": "~2.4.2",
    "@tarojs/components": "3.6.22",
    "@tarojs/helper": "3.6.22",
    "@tarojs/plugin-framework-react": "3.6.22",
    "@tarojs/plugin-platform-alipay": "3.6.22",
    "@tarojs/plugin-platform-h5": "3.6.22",
    "@tarojs/plugin-platform-jd": "3.6.22",
    "@tarojs/plugin-platform-qq": "3.6.22",
    "@tarojs/plugin-platform-swan": "3.6.22",
    "@tarojs/plugin-platform-tt": "3.6.22",
    "@tarojs/plugin-platform-weapp": "3.6.22",
    "@tarojs/react": "3.6.22",
    "@tarojs/rn-supporter": "3.6.22",
    "@tarojs/runtime": "3.6.22",
    "@tarojs/runtime-rn": "3.6.22",
    "@tarojs/shared": "3.6.22",
    "@tarojs/taro": "3.6.22",
    "@tarojs/taro-rn": "3.6.22",
    "expo": "~47.0.3",
    "expo-av": "~13.0.0",
    "expo-barcode-scanner": "~12.0.0",
    "expo-brightness": "~11.0.0",
    "expo-camera": "~13.0.0",
    "expo-file-system": "~15.1.1",
    "expo-image-picker": "~14.0.0",
    "expo-keep-awake": "~11.0.1",
    "expo-location": "~15.0.1",
    "expo-sensors": "~12.0.1",
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "react-native": "^0.70.1",
    "react-native-device-info": "~10.3.0",
    "react-native-gesture-handler": "~2.9.0",
    "react-native-pager-view": "~6.0.1",
    "react-native-safe-area-context": "~4.4.1",
    "react-native-screens": "~3.20.0",
    "react-native-svg": "~13.7.0",
    "react-native-syan-image-picker": "0.5.3",
    "react-native-webview": "~11.23.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.5",
    "@tarojs/cli": "3.6.22",
    "@tarojs/rn-runner": "3.6.22",
    "@tarojs/taro-loader": "3.6.22",
    "@tarojs/webpack5-runner": "3.6.22",
    "@types/node": "^18.15.11",
    "@types/react": "^18.0.0",
    "@types/react-native": "^0.69.8",
    "@types/webpack-env": "^1.13.6",
    "@typescript-eslint/eslint-plugin": "^5.20.0",
    "@typescript-eslint/parser": "^5.20.0",
    "babel-preset-taro": "3.6.22",
    "eslint": "^8.12.0",
    "eslint-config-taro": "3.6.22",
    "eslint-plugin-import": "^2.12.0",
    "eslint-plugin-react": "^7.8.2",
    "eslint-plugin-react-hooks": "^4.2.0",
    "install-peerdeps": "^3.0.3",
    "pod-install": "^0.1.23",
    "postcss": "^8.4.18",
    "react-refresh": "^0.11.0",
    "stylelint": "9.3.0",
    "ts-node": "^10.9.1",
    "typescript": "^4.1.0",
    "webpack": "^5.78.0"
  }
}

**Comment by Chen-jj** [comment]
RN 开发容易遇到各种环境问题，建议仔细阅读报错信息，使用关键词进行 Google 或 GPT 查询，这样能解决大部分问题。

1. 安装NDK

![image](https://github.com/NervJS/taro/assets/11807297/90658310-d22d-4017-9368-bf434d0f73ab)

2. 配置 compileSdkVersion

![image](https://github.com/NervJS/taro/assets/11807297/37b1f102-c288-4631-94ec-2ceb58204ebd)
